DTML.EduBot/Dialogs/RootDialog.cs

```diff
@@ -3,76 +3,265 @@
     using System;
     using System.Collections.Generic;
     using System.Collections.ObjectModel;
+    using System.Globalization;
     using System.Linq;
     using System.Threading.Tasks;
     using Autofac;
     using Microsoft.Bot.Builder.Dialogs;
     using Microsoft.Bot.Connector;
+    using DTML.EduBot.Common;
+    using DTML.EduBot.UserData;
 
     [Serializable]
     public class RootDialog : IDialog<string>
     {
+        private readonly int MAX_ATTEMPT = 2;
+
+        private const string Yes = "Yes";
+        private const string No = "No";
+
+        private static readonly IEnumerable<string> YesNoChoices = new ReadOnlyCollection<string>
+            (new List<String> {
+                Yes,
+                No});
+
         private const string ChatWithBot = "Chat With Bot";
-        private const string StartTheLessonPlan = "Start The Lesson Plan";
+        private const string StartTheLessonPlan = "Start English Lesson Plan";
 
-        private static readonly IEnumerable<string> Choices = new ReadOnlyCollection<string>
+        private static readonly IEnumerable<string> DialogChoices = new ReadOnlyCollection<string>
             (new List<String> {
                 ChatWithBot,
                 StartTheLessonPlan});
 
+        private const string TooManyAttemptMessage = "Sorry, you have attempted too many times :(";
+
         public Task StartAsync(IDialogContext context)
         {
             context.Wait(this.MessageReceivedAsync);
             return Task.CompletedTask;
         }
 
-        private Task MessageReceivedAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
+        private async Task MessageReceivedAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
         {
-            string friendlyUserName = context.Activity.From.Name;
+            var userText = (await result).Text;
             
+            string detectedLanguageIsoCode = await MessageTranslator.IdentifyLangAsync(userText);
+
+            using (var scope = WebApiApplication.FindContainer().BeginLifetimeScope())
+            {
+                UserData.UserData userData = new UserData.UserData();
+                userData.UserId = context.Activity.From.Id;
+                userData.NativeLanguageIsoCode = detectedLanguageIsoCode;
+
+                scope.Resolve<IUserDataRepository>().UpdateUserData(userData);
+
+                if (MessageTranslator.DEFAULT_LANGUAGE.Equals(detectedLanguageIsoCode))
+                {
+                    await context.PostAsync(BotPersonality.UserNameQuestion);
+
+                    // detected it's english language
+                    context.Wait(this.UserNameReceivedAsync);
+                }
+                else
```

Comment:
Offline discussed. The if/else statements is due to the scenario user replies in Spanish, but want to keep the conversation in English. Refactoring would require some time and thoughts.

In addition, do we want to refactor during the hackathon, or after?